security/s2a/internal/crypter/counter_test.go

```diff
@@ -0,0 +1,91 @@
+package crypter
+
+import (
+	"math"
+	"testing"
+)
+
+type counterTest struct {
+	desc          string
+	counter, want uint64
+	overflow      bool
+}
+
+func TestCounterInc(t *testing.T) {
+	for _, test := range []counterTest{
+		{
+			desc:    "basic 1",
+			counter: 0,
+			want:    1,
+		},
+		{
+			desc:    "basic 2",
+			counter: 123,
+			want:    124,
+		},
+		{
```

Comment:
Could you add a test vector with `counter: math.MaxUint64 - 1`, `want: math.MaxUint64` and verify that it does not overflow?

Reply:
Done. I had to add some extra conditions to check for this. I realized that my initial conditions didn't check for the overflow flag in the first call to `getAndIncrement`.